salt/modules/aptpkg.py

```diff
@@ -2305,6 +2308,16 @@ def mod_repo(repo, saltenv='base', **kwargs):
                 'Error: failed to add key from {0}'.format(key_url)
             )
 
+    elif 'key_text' in kwargs:
+        key_text = kwargs['key_text']
+        cmd = ['apt-key', 'add', '-']
+        out = __salt__['cmd.run_stdout'](cmd, stdin=key_text,
```

Comment:
@terminalmage Yes I've tested it with a local setup and it works (though if others want to verify that would be good). I looked at the existing tests for this area of the code and they all looked rather superficial (ie. testing of mocked output, not functionality), so I did not add any tests for this use case. If you have any suggestions for implemented a test for the new *functionality* I'm happy to have a look.

The use of `-` is not a shell-ism. From `man apt-key`
> The key is read from the filename given with the parameter filename or if the filename is - from standard input.

There is absolutely nothing wrong with being cautious ;)